configure.ac

```diff
@@ -236,7 +236,14 @@ AC_CHECK_PROG(YASM_CHECK, yasm, yes)
 if test x"$YASM_CHECK" = x"yes"; then
   if yasm -f elf64 src/common/crc32c_intel_fast_asm.S -o /dev/null; then
      echo 'we have a modern and working yasm'
-     if test `arch` = "x86_64" ; then
+     # FreeBSD has uname -p not arch(1), so do not exec arch
+     arch_x86_64=0
+     if test `uname -p` = "amd64"; then
+	arch_x86_64=1
+     elif test  -x /usr/bin/arch -a `arch` = "x86_64" ; then
```

Comment:
why you'd like to check a specific path for `arch` and call it just using "arch" (i.e. without specifying the full path?